sklearn/model_selection/_split.py

```diff
@@ -536,6 +536,32 @@ def _iter_test_indices(self, X, y, groups):
         for f in range(self.n_splits):
             yield np.where(indices == f)[0]
 
+    def split(self, X, y=None, groups=None):
+        """Generate indices to split data into training and test set.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data, where n_samples is the number of samples
+            and n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
```

Comment:
y is optional, groups is not, right?

Reply:
You're right, actually I believe it is completely ignored. Other methods in this file have "Always ignored, exists for compatibility." in their docstring for `y` or `groups` so I'll add that.

Reply:
Well, actually it isn't ignored in this function since it throws an error if `y` has inconsistent length. However as far as I can tell `y` does not change the result.

Reply:
cool. Yeah there's no grouped stratified splitter so far, iirc.